frontend/src/ToDoGallery.tsx

```diff
@@ -0,0 +1,81 @@
+import React, {useEffect, useState} from 'react';
+import {status, ToDo} from "./models/model";
+import ToDoItem from "./ToDoItem";
+
+
+export default function ToDoGallery() {
+
+    const [toDoItems, setToDoItems] = useState([] as Array<ToDo>)
+    const [id, setId] = useState('')
+    const [task, setTask] = useState('')
+    const [description, setDescription] = useState('')
+    const [stati, setStati] = useState(status.Open)
+
+    const requestBody = {
+        task: task,
+        description: description
+    }
+
+    const url = "http://localhost:8080/todos";
+
+    useEffect(() => {
+
+        fetch(url)
+            .then(response => response.json())
+            .then((data) => {
+                setToDoItems(data)
+            })
+    }, []);
+
+
+    const getToDoGallery = () => {
+        fetch(url)
+            .then(response => response.json())
+            .then(items => setToDoItems(items))
+    }
+
+    const setAttributes = () => {
+        fetch(url, {
+            method: 'POST',
+            body: JSON.stringify(requestBody),
+            headers: {'Content-Type': 'application/json'}
+        })
+            .then(() => getToDoGallery())
+            .then(() => {
+                setTask('')
+                setDescription('')
+                setStati(status.Open)
+            })
+    }
+
+    const deleteById = () => {
+        fetch(url+id, {
+            method: 'DELETE',
+            body: JSON.stringify(requestBody),
+            headers: {
+                'Content-Type': 'application/json'
+            }, })
+            .then((response) => response.json())
+            .then((data) => setToDoItems(data))
+    }
+
+    return (
+        <div>
+
+            <div className="text fields">
+                <input type="text" placeholder="Task" value={task} onChange={a => setTask(a.target.value)}/>
+                <input type="text" placeholder="description" value={description} onChange={a => setDescription(a.target.value)}/>
+                <input type="checkbox"/>
+                <input type="text" placeholder="id" value={id} onChange={a => setId(a.target.value)}/>
```

Comment:
Statt eines ID Feldes hier und des delete-Buttons weiter unten, würde ich den Button zum Löschen in die Komponente packen, die ein Todo darstellt. Da hat man dann ja auch die ID an der Hand. Der Nutzer kennt die IDs ja normalerweise gar nicht.